grpc/src/main/java/io/stargate/grpc/service/streaming/StreamingExceptionHandler.java

```diff
@@ -0,0 +1,54 @@
+package io.stargate.grpc.service.streaming;
+
+import com.google.protobuf.Any;
+import com.google.rpc.ErrorInfo;
+import io.grpc.Metadata;
+import io.grpc.Status;
+import io.grpc.StatusException;
+import io.grpc.StatusRuntimeException;
+import io.stargate.grpc.service.ExceptionHandler;
+import io.stargate.grpc.service.SuccessHandler;
+import io.stargate.proto.QueryOuterClass;
+import java.util.Optional;
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+public class StreamingExceptionHandler extends ExceptionHandler {
+  private final SuccessHandler successHandler;
+
+  public StreamingExceptionHandler(SuccessHandler successHandler) {
+    this.successHandler = successHandler;
+  }
+
+  @Override
+  protected void onError(
+      @Nullable Status status, @Nonnull Throwable throwable, @Nullable Metadata trailer) {
+    // propagate streaming error as a Status
+    successHandler.handleResponse(
+        QueryOuterClass.StreamingResponse.newBuilder()
+            .setStatus(convertStatus(status, throwable))
+            .build());
+  }
+
+  private com.google.rpc.Status convertStatus(
```

Comment:
how this differ from the other exception handler and is this something we can contribute to the abstract class?